submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/service/SubmissionRepository.kt

```diff
@@ -42,9 +52,30 @@ class SubmissionRepository(
             .map { it.asSimpleSubmission() }
     }
 
-    private fun getSubmission(accNo: String) =
-        submissionRepository.getByAccNo(accNo) ?: throw SubmissionNotFoundException(accNo)
+    /**
+     * Load submission information strategy used is basically first load submission and then load each section and it
+     * subsections recursively starting from the root section.
+     */
+    private fun submission(accNo: String, version: Int? = null): DbSubmission {
+        logger.info { "loading submission $accNo" }
+
+        val submission = findSubmission(accNo, version)
+        submission ?: throw SubmissionNotFoundException(accNo)
+        loadSection(submission.rootSectionId)
+
+        logger.info { "Loaded submission $accNo" }
+        return submission
+    }
+
+    private fun findSubmission(accNo: String, version: Int?): DbSubmission? {
+        return when (version) {
```

Comment:
Are we sure we need this logic? why not just leave the version with a default value?

Reply:
I notice that was kind of misleading. With default parameter, we do not know if we need a specific version or we need an active one